sky/backends/backend_utils.py

```diff
@@ -3633,6 +3633,29 @@ def check_rsync_installed() -> None:
                 '  $ sudo apt install rsync') from None
 
 
+def check_stale_runtime_on_remote(returncode: int, stderr: str,
+                                  cluster_name: str) -> None:
+    """Raises RuntimeError if remote SkyPilot runtime needs to be updated.
+
+    We detect this by parsing certain backward-incompatible error messages from
+    `stderr`. Typically due to the local client version just got updated, and
+    the remote runtime is an older version.
+    """
+    pattern = re.compile(r'AttributeError: module \'sky\.(.*)\' has no '
+                         r'attribute \'(.*)\'')
+    if returncode != 0:
+        attribute_error = re.findall(pattern, stderr)
+        if attribute_error:
```

Comment:
I think we should keep the `'SkyPilot runtime is too old' in stderr` instead of `attribute_error`? The attribute_error is to handle the case when our skylet versioning was not introduced?